### PR TITLE
Refactor to MirageJS

### PR DESCRIPTION
**Pull request by minkimcello**
## Motivation
Screen Shot 2020-07-14 at 4 57 15 PM
This is part of an on-going initiative to completely migrate all of `folio-org` projects from `@bigtest/mirage` to `miragejs`.

[`@bigtest/mirage`](https://github.com/bigtestjs/mirage/) was originally a prototype to make `mirage` available to any framework, but it has since been archived in favor of [`miragejs`](https://miragejs.com/). The documentation for `miragejs` is not only prodigious for its extensive API docs but also for its tutorials and guides. Furthermore, it receives regular bug-fixes and maintenance releases in thanks to its extremely large and active community.

Previous PR: [@folio-org/ui-search #232](https://github.com/folio-org/ui-search/pull/232)
_Linking the previous pull request so that subsequent ones can be found tagged below._

<details>
<summary>Overview of progress at the time of the creation of this PR (in chronological order):</summary>
<br>
✅ <code>ui-checkin</code>
<br>
✅ <code>ui-checkout</code>
<br>
☑️ <code>ui-circulation</code>
<br>
☑️ <code>ui-calendar</code>
<br>
✅ <code>ui-agreements</code>
<br>
☑️ <code>ui-app-template</code>
<br>
☑️ <code>ui-inventory</code>
<br>
☑️ <code>ui-data-export</code>
<br>
✅ <code>ui-finc-config</code>
<br>
☑️ <code>ui-licenses</code>
<br>
☑️ <code>ui-erm-comparisons</code>
<br>
☑️ <code>ui-myprofile</code>
<br>
☑️ <code>ui-local-kb-admin</code>
<br>
☑️ <code>ui-marccat</code>
<br>
☑️ <code>ui-finc-select</code>
<br>
☑️ <code>ui-erm-usage</code>
<br>
☑️ <code>ui-notes</code>
<br>
☑️ <code>ui-oai-pmh</code>
<br>
☑️ <code>ui-plugin-find-user</code>
<br>
☑️ <code>ui-plugin-find-eresource</code>
<br>
☑️ <code>ui-plugin-find-package-title</code>
<br>
☑️ <code>ui-plugin-find-import-profile</code>
<br>
☑️ <code>ui-requests</code>
<br>
☑️ <code>ui-tenant-settings</code>
</details>

## Approach
- [x] Cloned, installed, and then ran tests to double check that the tests are passing before refactoring.
- [x] Removed `@bigtest/mirage` and added `inflected`, `faker`, and `miragejs`.
- [x] ESlint & ran tests after the refactor to confirm they all pass.
  - [x] :warning: I try to focus on the `miragejs` refactor but I got carried away and fixed all the promises in the test suites. If you don't return promises, the results of the tests are inconsistent because of the race condition.
- [x] Updated CHANGELOG.

### CODE REVIEW

File: test/bigtest/tests/service-point-create-test.js [hunk 1]
```diff
@@ -10,13 +10,7 @@ describe('ServicePointCreate', () => {
 
   beforeEach(function () {
     servicePoint = this.server.create('servicePoint');
-  });
-
-  beforeEach(function () {
     this.visit(`/settings/tenant-settings/servicePoints/${servicePoint.id}?layer=edit`);
```
Comment (VictorSoroka1) [inline test/bigtest/tests/service-point-create-test.js:13]: Can this be removed at all?

File: test/bigtest/tests/location-libraries-test.js [hunk 2]
```diff
@@ -20,7 +20,7 @@ describe('LocationLibraries', () => {
   describe('Institution is present', () => {
     beforeEach(function () {
       institution = this.server.create('institution');
-      this.visit('/settings/tenant-settings/location-libraries');
+      return this.visit('/settings/tenant-settings/location-libraries');
```
Comment (VictorSoroka1) [inline test/bigtest/tests/location-libraries-test.js:23]: Is `return` is really needed?

File: test/bigtest/network/config.js [hunk 3]
```diff
@@ -1,5 +1,10 @@
-// typical mirage config export
+import { inflections } from 'inflected';
+
 export default function config() {
+  inflections('en', function (inflect) {
+    inflect.irregular('campus', 'campus');
```
Comment (VictorSoroka1) [inline test/bigtest/network/config.js:5]: Please add comment for the logic behind this.

Comment (minkimcello) [reply test/bigtest/network/config.js:6]: Done.

File: test/bigtest/tests/service-point-create-test.js (hunk 1 above)
Comment (minkimcello) [reply test/bigtest/tests/service-point-create-test.js:13]: Yes. Both [`server.create`](https://github.com/miragejs/miragejs/blob/master/lib/server.js#L861) and [`this.visit`](https://github.com/bigtestjs/react/blob/master/src/visit.js#L20) are synchronous so we can combine the `beforeEach`.

File: test/bigtest/tests/location-libraries-test.js (hunk 2 above)
Comment (minkimcello) [reply test/bigtest/tests/location-libraries-test.js]: Oh this was my mistake. I thought they were asynchronous and believed that was the cause of the inconsistency but those methods are synchronous. I undid the `return`.

File: test/bigtest/tests/service-point-create-test.js (hunk 1 above)
Comment (VictorSoroka1) [reply test/bigtest/tests/service-point-create-test.js:13]: I mean that there two consecutive this.visit calls. As I understand only the last one is needed.

Comment (minkimcello) [reply test/bigtest/tests/service-point-create-test.js:13]: Oh, I mean that was there before and doesn't relate to the mirage refactor. We'll reevaluate the tests on a different PR.

Comment (VictorSoroka1) [reply test/bigtest/tests/service-point-create-test.js:13]: Sounds goog.

**Pull request merged**